### DISCUSSION TITLE
Reusing the same Popover for mutliple element

### DISCUSSION THREAD

**Discussion by agupta1989** [chakra-ui/chakra-ui#8553] [Q&A] [answered]
Hi,

In my application, there are more than 10 list items. On each list item click I need to show a popover using portal. A simple solution was developed, where each list item have Popover component where item is set as Popover trigger. All good till here, things are working as expected.

but then I noticed something in the browser that, for each instance, a new instance of portal and popover component is getting created !

![image](https://github.com/chakra-ui/chakra-ui/assets/3787965/c52d723d-2077-4bf4-98c1-946a98249531)

This is something I feel is not optimized unless I am overthinking it. Can somebody help me understand whether its ok or is there any optimization technique we can use, where I can use the same Portal + Popover combination for all the list item ?

I checked the document, the trigger needs to be within the popover component. Is there anywhere to keep the trigger outside while reusing the same Popover component in my application ?

**Comment by segunadebayo** [comment]
Use one `Popover.Root`. Point it at the current element with `positioning.getAnchorRect`.

Do not mount a Popover per trigger if only the data changes.

**Comment by segunadebayo** [comment, marked as answer]
Multiple triggers shipped. Put `value` on each `Popover.Trigger` under one `Root`.

https://chakra-ui.com/docs/components/popover#multiple-triggers

`getAnchorRect` is still the right tool when the target is not a trigger.

**Reply by agupta1989** [reply]
> Multiple triggers shipped. Put `value` on each `Popover.Trigger` under one `Root`.
> 
> https://chakra-ui.com/docs/components/popover#multiple-triggers
> 
> `getAnchorRect` is still the right tool when the target is not a trigger.

Thanks !

Actually the issue / question I reported was for Chakra ui v2. but anyways since we are also migrating to chakra ui v3 (thanks to the agentic skills you guys have shared, made life lot easier) we will try to use the approach suggested.
